python_legacy/tests/api/expressions/conftest.py

```diff
@@ -0,0 +1,425 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from decimal import Decimal
+import io
+import pickle
+import time
+import uuid
+
+from iceberg.api import DataFile, ManifestFile, PartitionFieldSummary, PartitionSpec
+from iceberg.api.expressions import (BoundPredicate,
+                                     Expressions,
+                                     ExpressionVisitors,
+                                     Literal,
+                                     Operation,
+                                     UnboundPredicate)
+from iceberg.api.schema import Schema
+from iceberg.api.struct_like import StructLike
+from iceberg.api.types import (BinaryType,
+                               Conversions,
+                               DateType,
+                               DecimalType,
+                               FixedType,
+                               IntegerType,
+                               NestedField,
+                               StringType,
+                               TimestampType,
+                               TimeType)
+import pytest
+
+exp_schema = Schema(NestedField.optional(34, "a", IntegerType.get()))
+
+
+class TestHelpers(object):
+
+    @staticmethod
+    def assert_all_references_bound(message, expr):
+        ExpressionVisitors.visit(expr, TestHelpers.CheckReferencesBound(message))
+
+    @staticmethod
+    def assert_and_unwrap(expr, expected=None):
+        if expected is not None:
+            assert isinstance(expr, expected)
+        else:
+            assert isinstance(expr, BoundPredicate)
+
+        return expr
+
+    @staticmethod
+    def round_trip_serialize(type_var):
+        stream = io.BytesIO()
+        pickle.dump(type_var, stream, pickle.HIGHEST_PROTOCOL)
+        stream.seek(0)
+
+        return pickle.load(stream)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Pickle and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue. https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b301-pickle